src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -153,6 +148,54 @@ private static void loadStyleMaps()
         }
     }
 
+
+    /**
+     * load the schematics from the jar
+     */
+    private static void loadStyleMapsJar()
+    {
+        URI uri = null;
+        try
+        {
+            Log.getLogger().info("loadStyleMaps jar: " + ColonyManager.class.getResource(SCHEMATICS_ASSET_PATH).toURI()); 
+            uri = ColonyManager.class.getResource(SCHEMATICS_ASSET_PATH).toURI();
+        }
+        catch (@NotNull URISyntaxException e)
+        {
+            Log.getLogger().error("loadStyleMaps : ",e); 
+            return;
+        }
+
+        if ("jar".equals(uri.getScheme()))
+        {
+            try (FileSystem fileSystem = FileSystems.getFileSystem(ColonyManager.class.getResource(SCHEMATICS_ASSET_PATH).toURI()))
+            {
+                final Path basePath = fileSystem.getPath(SCHEMATICS_ASSET_PATH);
+                Log.getLogger().info("Load huts or decorations from jar");
+                loadSchematicsForPrefix(basePath, SCHEMATICS_PREFIX);
+            }
+            catch (@NotNull IOException | URISyntaxException | FileSystemNotFoundException e1)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)